pgrx/src/tupdesc.rs

```diff
@@ -174,13 +175,11 @@ impl<'a> PgTupleDesc<'a> {
     ```
     */
     pub fn for_composite_type(name: &str) -> Option<PgTupleDesc<'a>> {
-        unsafe {
-            let mut typoid = pg_sys::Oid::INVALID;
-            let mut typmod = 0;
-            pg_sys::parseTypeString(name.as_pg_cstr(), &mut typoid, &mut typmod, true);
+        let typoid = PgTryBuilder::new(|| Some(regtypein(name)))
+            .catch_when(PgSqlErrorCode::ERRCODE_UNDEFINED_OBJECT, |_| None)
+            .execute()?;
```

Comment:
Huh, what's the difference?

Reply:
Ah I see, previously we were saying `missing_ok = true` and letting an `InvalidOid` potentially pass through, which would become a None. But `parseTypeString` in new Postgres versions now has an `ErrorSaveContext*`, and we don't have an API design that supports using that instead for error handling. So you chose to switch to regtypein, which does that business for us, and we do throw-catch based errors instead, and this gives us something consistent across all our supported Postgres versions with effectively equivalent functionality. We might want to add ErrorSaveContext support, sounds kinda Rusty to do, but... later.

Reply:
Exactly.  

Also, in looking at it in detail, I think that `parseTypeString` was probably the wrong thing to be using in the first place, but it just happened to work out okay.  Using the `regtypein` function I think would actually be the preferred way of doing this.